### DISCUSSION TITLE
Difference between union and union_instantiations (for Analysis)

### DISCUSSION THREAD

**Discussion by skius** [egraphs-good/egg#120] [Q&A] [answered]
Hi!

I was going through the Analysis examples and came across a small difference between [`tests/prop.rs`](https://github.com/egraphs-good/egg/blob/main/tests/prop.rs#L50)  and the [`Analysis` docs](https://github.com/egraphs-good/egg/blob/main/src/language.rs#L545): The former uses `union_instantiations` while the latter simply uses `union`. 

What is the difference between the two? I know that `union_instantiations` doesn't exist in 0.6.0, but is there e.g. a performance or expressibility benefit to it (apart from constructing Explanations)?

**Comment by mwillsey** [comment, marked as answer]
Hi there! We still to working on the explanations feature, so maybe @oflatt will want to say more, but here is my current understanding.

You can use `union` if you want, but if you then ask for explanations, you won't get one for those unions. Using `union_instantiations` will work provide that explanation if they are enabled, but [behave like union otherwise](https://github.com/egraphs-good/egg/blob/18d7f616ad6412526aff825876df4770e04418f2/src/egraph.rs#L464).

**Reply by oflatt** [reply]
That's right! In fact, `union` is disallowed when explanations are turned on.
